src/main/scala/is/hail/methods/Aggregators.scala

```diff
@@ -17,6 +17,75 @@ import scala.reflect.ClassTag
 
 object Aggregators {
 
+  def buildVariantAggregationsByKey(vsm: VariantSampleMatrix, nKeys: Int, keyMap: Array[Int], ec: EvalContext): Option[(RegionValue) => Iterator[() => Unit]] =
+    buildVariantAggregationsByKey(vsm.sparkContext, vsm.matrixType, vsm.value.localValue, nKeys, keyMap, ec)
+
+  // keyMap is just a mapping of sampleIds.map { s => newKey(s) }
+  def buildVariantAggregationsByKey(sc: SparkContext,
+    typ: MatrixType,
+    localValue: VSMLocalValue,
+    nKeys: Int,
+    keyMap: Array[Int],
+    ec: EvalContext): Option[(RegionValue) => Iterator[() => Unit]] = {
+
+    val aggregations = ec.aggregations
+    if (aggregations.isEmpty)
```

Comment:
this optimization might introduce too much additional complexity, see other comments